src/tcp.toit

```diff
@@ -31,3 +93,54 @@ class TcpTransport implements Transport:
       with_timeout timeout:
         return Packet.deserialize reader_
     return null
+
+  reconnect:
+    // TODO(florian): implement retries and exponential back-off.
+    old_socket := socket_
+    reconnecting_mutex = monitor.Mutex
+    reconnecting_mutex.do:
+      if not identical old_socket socket_: return
+      if old_socket: old_socket.close
+
+      socket := new_connection_
+      writer_ = writer.Writer socket
+      reader_ = reader.BufferedReader socket
+      // Send messages immediately.
+      socket.set_no_delay true
+
+      // Set the new socket_ at the very end. This way we will try to
+      // reconnect again if we are interrupted by a timeout.
+      socket_ = socket
+
+  new_connection_ -> tcp.Socket:
+    return interface_.tcp_connect host_ port_
+
+  close:
+    if socket_: socket_.close
+    socket_ = null
+    writer_ = null
+    reader_ = null
+
+
+class ReconnectingTlsTcpTransport_ extends ReconnectingTcpTransport_:
```

Comment:
Maybe dro the Tcp part of the name? It is a local name and it doesn't add much to have it here.